### PR TITLE
creds: fix cat with encrypted credentials

### PR DESCRIPTION
**Pull request by SimonPilkington**
See #34547

### CODE REVIEW

File: src/creds/creds.c [hunk 1]
```diff
@@ -427,6 +427,9 @@ static int verb_cat(int argc, char **argv, void *userdata) {
                 /* Look both in regular and in encrypted credentials */
                 for (encrypted = 0; encrypted < 2; encrypted++) {
                         _cleanup_closedir_ DIR *d = NULL;
+                        ReadFullFileFlags flags = READ_FULL_FILE_SECURE|READ_FULL_FILE_WARN_WORLD_READABLE;
```
Comment (YHNdnzj) [inline src/creds/creds.c]: I'd place this right before the read_full_file call

**Review by YHNdnzj — APPROVED** [review]
LGTM, but one nit.

Please also link to the issue in the commit msg.

**Comment by SimonPilkington** [comment]
Added issue to commit message.

File: src/creds/creds.c (hunk 1 above)
Comment (YHNdnzj) [reply src/creds/creds.c]: (I mean the definition of the flags var, too)

Comment (SimonPilkington) [reply src/creds/creds.c]: My mistake. From a cursory look it seemed like C89 declarations were being used.

**Comment by SimonPilkington** [comment]
Tests are failing now because they're `base64 -d`ing the output of `systemd-creds encrypt`. 
https://github.com/systemd/systemd/blob/df4d09f78eed6daef508a6a0421759b3c6767256/test/units/TEST-54-CREDS.sh#L46

That explains how they were passing before but doing  this doesn't make much sense to me. @YHNdnzj, thoughts?

File: NEWS [hunk 2]
```diff
@@ -11,7 +11,13 @@ CHANGES WITH 257 in spe:
           be updated accordingly. This change has been made to make it harder
           to accidentally delete too many files when using --purge incorrectly.
 
-        Announcements of Future Feature Removals and Incompatible Changes:
+        * The systemd-creds 'cat' verb could previously only read encrypted
+          credentials if they were raw binary data, making it inconsistent
+          with the 'decrypt' verb and and the LoadCredentialEncrypted= service
+          setting, which expect base64-encoded data. This is now fixed,
```
Comment (YHNdnzj) [inline NEWS]: I'd always mention the new behavior first, and then describe the previous (wrong) behavior. Also, "potentially breaking existing setups" is redundant.

**Review by YHNdnzj — APPROVED** [review]

**Comment by yuwata** [comment]
Please split out the NEWS change into another commit. Then, we can easily backport the fix.

File: NEWS (hunk 2 above)
Comment (SimonPilkington) [reply NEWS]: How's this?

File: NEWS [hunk 3]
```diff
@@ -11,7 +11,12 @@ CHANGES WITH 257 in spe:
           be updated accordingly. This change has been made to make it harder
           to accidentally delete too many files when using --purge incorrectly.
 
-        Announcements of Future Feature Removals and Incompatible Changes:
+        * The systemd-creds 'cat' verb new expects base64-encoded encrypted
```
Comment (cvlc12) [inline NEWS]:
```suggestion
        * The systemd-creds 'cat' verb now expects base64-encoded encrypted
```
new -> now if I'm not mistaken ?

Comment (SimonPilkington) [reply NEWS]: Yes, thanks.

File: NEWS (hunk 2 above)
Comment (SimonPilkington) [reply NEWS]: Assuming silence == approval, unresolve if not.

**Pull request merged**